### PR TITLE
[APPSEC-9936] Update AppSec response content_type resolution

### PR DESCRIPTION
**Pull request by GustavoCaso**
**What does this PR do?**

Based on the [internal] specs. The default content type and the response of the blocked page should be `JSON`. 

I update the code that handles that, I also added a bunch os new specs. I fixed an issue in which spaces between the `,` for separating multiple HTTP accept values.

**Motivation**

**Additional Notes**

**How to test the change?**

CI

### CODE REVIEW

File: lib/datadog/appsec/response.rb [hunk 1]
```diff
@@ -43,20 +43,22 @@ def negotiate(env)
...
           return DEFAULT_CONTENT_TYPE unless env.key?('HTTP_ACCEPT')
 
-          accepted = env['HTTP_ACCEPT'].split(',').map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
+          accept_types = env['HTTP_ACCEPT'].split(',').map(&:strip)
+
+          accepted = accept_types.map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
 
-          accepted.each_with_object(DEFAULT_CONTENT_TYPE) do |range, _default|
-            match = FORMAT_MAP.keys.find { |type| range === type }
+          accepted.each do |range|
```
Comment (lloeki) [inline lib/datadog/appsec/response.rb:60]: `accepted.each_with_object(DEFAULT_CONTENT_TYPE)` was making it so it returns `DEFAULT_CONTENT_TYPE` when no match has occured.

`accepted.each` now instead returns `accepted`, which is an array of `MediaRange`, so:

- the return value of `content_type` is now inconsistent: when a match is found it returns a content type but when none is found it returns an array of `MediaRange`
- when there is no match it does not return the default value anymore, which will break `FORMAT_MAP[content_type]`

File: lib/datadog/appsec/response.rb [hunk 2]
```diff
@@ -43,20 +43,22 @@ def negotiate(env)
...
+            match = FORMAT_MAP.keys.find { |key| range === key }
```
Comment (lloeki) [inline lib/datadog/appsec/response.rb:61]: `FORMAT_MAP` keys are content types, so I'm questioning the name change here.

I would even change it that way to make it clearer:

```
type_match = FORMAT_MAP.keys.find { |type| range === type }

return type_match if type_match
```

**Review by lloeki — CHANGES_REQUESTED** [review]
Overall OK to change the default, but I believe there's a bug introduced here.

File: lib/datadog/appsec/response.rb [hunk 3]
```diff
@@ -36,30 +36,34 @@ def negotiate(env)
...
-          accepted = env['HTTP_ACCEPT'].split(',').map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
+          accept_types = env['HTTP_ACCEPT'].split(',').map(&:strip)
 
-          accepted.each_with_object(DEFAULT_CONTENT_TYPE) do |range, _default|
-            match = FORMAT_MAP.keys.find { |type| range === type }
+          accepted = accept_types.map { |m| Utils::HTTP::MediaRange.new(m) }.sort!.reverse!
 
-            return match if match
+          accepted.each do |range|
+            type_match = CONTENT_TYPE_TO_FORMAT.keys.find { |type| range === type }
```
Comment (ivoanjo) [inline lib/datadog/appsec/response.rb:61]: I noticed `MediaRange#===` with a `String` as input will parse that input into a `MediaRange` before doing anything else. Aka, we keep parsing the items in `CONTENT_TYPE_TO_FORMAT` all the type on this check. Maybe something to improve?

**Review by ivoanjo — APPROVED** [review]
LGTM :+1:

File: lib/datadog/appsec/response.rb (hunk 3 above)
Comment (GustavoCaso) [reply lib/datadog/appsec/response.rb:61]: Sounds good 😄

**Pull request merged**